addons/website_slides/static/src/js/public/components/slide_upload_dialog/slide_upload_category.js

```diff
@@ -0,0 +1,431 @@
+     * field or the 'document_google_url' / 'image_google_url' fields respectively.
+     */
+    onClickSourceType(isLocalSource) {
+        this.state.form.isLocalSource = isLocalSource;
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    // Alert messages
+
+    /**
+     * @param {string} message
+     */
+    _alertDisplay(message, alertClass = "alert-warning") {
+        this.state.alert.message = message;
+        this.state.alert.class = alertClass;
+        this.state.alert.show = true;
+    }
+
+    _alertRemove() {
+        this.state.alert.show = false;
+        this.state.alert.message = "";
+        this.state.alert.class = "";
+    }
+
+    // Category and tag SelectMenus
+
+    /**
+     * Get value for category_id and tag_ids (ORM cmd) to send to server
+     */
+    _getSelectMenuValues() {
+        const result = {};
+        // tags
+        if (this.state.choices.tagIds.length > 0) {
+            result["tag_ids"] = this.state.choices.tagIds.map((tagId) => {
+                const tag = this.state.choices.tags.find((tag) => tag.value === tagId);
+
+                if (this._toCreate(tag.value)) {
+                    return [0, 0, { name: tag.label }];
+                } else {
+                    return [4, tag.value];
+                }
+            });
+        }
+        // category
+        if (!this.defaultCategoryId) {
+            if (this._toCreate(this.state.choices.categoryId)) {
+                const category = this.state.choices.categories.find((cat) => {
+                    return cat.value === this.state.choices.categoryId;
+                });
+                result["category_id"] = [0, { name: category.label }];
+            } else {
+                const categoryId = parseInt(this.state.choices.categoryId, 10);
+                result["category_id"] = [categoryId];
+                this.categoryID = categoryId;
+            }
+        } else {
+            result["category_id"] = [this.defaultCategoryId];
+        }
+        return result;
+    }
+
+    /**
+     * Fetch available course categories and tags
+     */
+    async _fetch_choices(type, domain = [], fields = ["name"]) {
+        const choices = [];
+        const results = await this.rpc(`/slides/${type}/search_read`, { fields, domain });
+
+        results.read_results.forEach((choice) => {
+            choices.push({ value: choice.id, label: choice.name });
+        });
+        return choices;
+    }
+
+    /**
+     * Check whether it is a new category/tag or not
+     */
+    _toCreate(value) {
+        return typeof value === "string" && value.startsWith("temp");
+    }
+
+    // Form
+
+    _fileReset() {
+        document.getElementById("upload").value = "";
+        this.file.name = false;
+    }
+
+    _formValidate() {
```

Comment:
Unless it is used elsewhere, I wouldn't keep this separate from where it's called from.
Also does `wasValidated` mean that validity was confirmed? or rather that it was checked?